### PR TITLE
Modified checking for null context EObject in check file.

### PR DESCRIPTION
**Pull request by KrisLimbo**
If issue expression is equal to the implicit/input variable,
no null checking for context EObject is performed.
Otherwise, perform null checking and log context information
for debugging null EObject context object.

### CODE REVIEW

**Review by andrewL-avlq — APPROVED** [review]

File: com.avaloq.tools.ddk.check.core/src/com/avaloq/tools/ddk/check/generator/CheckCompiler.java [hunk 1]
```diff
@@ -264,14 +264,21 @@ protected void _toJavaStatement(final XIssueExpression expr, final ITreeAppendab
...
     }
 
-    // checking for null context EObject
-    b.append("if (");
+    boolean issueExpressionEqualsImplicitVariable = true;
     if (markerObject != null) {
-      internalToConvertedExpression(markerObject, b, getLightweightType(eObjectType));
-    } else {
-      b.append(getContextImplicitVariableName(expr));
+      issueExpressionEqualsImplicitVariable = markerObject.toString().equalsIgnoreCase(getContextImplicitVariableName(expr));
```
Comment (gregdyke) [inline com.avaloq.tools.ddk.check.core/src/com/avaloq/tools/ddk/check/generator/CheckCompiler.java:269]: I don't think equalsIgnoreCase is what we want here (but can fix this in a followup)

File: com.avaloq.tools.ddk.check.core/src/com/avaloq/tools/ddk/check/generator/CheckCompiler.java [hunk 2]
```diff
@@ -264,14 +264,21 @@ protected void _toJavaStatement(final XIssueExpression expr, final ITreeAppendab
...
+    }
+
+    if (!issueExpressionEqualsImplicitVariable) {
+      // checking for null context EObject
+      b.append("if (");
+      if (markerObject != null) {
+        internalToConvertedExpression(markerObject, b, getLightweightType(eObjectType));
+      } else {
+        b.append(getContextImplicitVariableName(expr));
```
Comment (gregdyke) [inline com.avaloq.tools.ddk.check.core/src/com/avaloq/tools/ddk/check/generator/CheckCompiler.java:278]: this no longer needs to be here?

Comment (gregdyke) [reply com.avaloq.tools.ddk.check.core/src/com/avaloq/tools/ddk/check/generator/CheckCompiler.java:278]: specifically the else, as we don't need to test if contextImplicitVariable is null

Comment (gregdyke) [reply com.avaloq.tools.ddk.check.core/src/com/avaloq/tools/ddk/check/generator/CheckCompiler.java:278]: I take it back, we don't need the whole if at line 275, based on what we already have at line 268

Comment (KrisLimbo) [reply com.avaloq.tools.ddk.check.core/src/com/avaloq/tools/ddk/check/generator/CheckCompiler.java:278]: Thank you for the comment. It is redundant.

File: com.avaloq.tools.ddk.check.core/src/com/avaloq/tools/ddk/check/generator/CheckCompiler.java (hunk 1 above)
Comment (KrisLimbo) [reply com.avaloq.tools.ddk.check.core/src/com/avaloq/tools/ddk/check/generator/CheckCompiler.java:269]: Done.

File: com.avaloq.tools.ddk.check.core/src/com/avaloq/tools/ddk/check/generator/CheckCompiler.java (hunk 2 above)
Comment (KrisLimbo) [reply com.avaloq.tools.ddk.check.core/src/com/avaloq/tools/ddk/check/generator/CheckCompiler.java:278]: Done.
